crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -793,16 +799,25 @@ where
             EngineSyncEvent::PipelineFinished { result, reached_max_block } => {
                 match result {
                     Ok(ctrl) => {
-                        if ctrl.is_unwind() {
-                            self.sync.set_pipeline_sync_target(current_state.head_block_hash);
-                        } else if reached_max_block {
+                        if reached_max_block {
                             // Terminate the sync early if it's reached the maximum user
                             // configured block.
                             return Some(Ok(()))
                         }
 
+                        let Some(current_state) = self.forkchoice_state else {
+                            warn!(target: "consensus::engine", "No forkchoice state available");
+                            return None
```

Comment:
this branch is only possible if launched via debug tip and no FCU

can we add this info here?